lib/download.sh

```diff
@@ -0,0 +1,128 @@
+#
+# this expects lib/common.sh to be sourced
+
+wget_and_verify()
+{
+    local url="${1:?url missing}"
+    local sha256="${2:?sha256 missing}"
+    local target="${3:?target missing}"
+    local checksum
+
+    declare -a args=(
+        --no-verbose
+        -O "${target}"
+        "${url}"
+    )
+
+    wget "${args[@]}"
+
+    checksum="$(sha256sum "${target}" | awk '{print $1;}')"
+    if [[ "${checksum}" != "${sha256}" ]]; then
+        echo >&2 "fatal: ${url} binary checksum '${checksum}' differs from expected checksum '${sha256}'"
+        return 1
+    fi
+
+    return 0
+}
+
+download_and_install_krew()
+{
+    local tmp_dir
+    tmp_dir="$(mktemp -d)"
+    pushd "${tmp_dir}" || return 1
+
+    KERNEL_OS="$(uname | tr '[:upper:]' '[:lower:]')"
+    ARCH="$(uname -m | sed -e 's/x86_64/amd64/' -e 's/\(arm\)\(64\)\?.*/\1\2/' -e 's/aarch64$/arm64/')"
+    KREW="krew-${KERNEL_OS}_${ARCH}"
+    # TODO: can't pin latest
+    KREW_URL="https://github.com/kubernetes-sigs/krew/releases/download/${KREW_VERSION}/${KREW}.tar.gz"
+    wget_and_verify "${KREW_URL}" "${KREW_SHA256}" "${KREW}.tar.gz"
+    tar zxvf "${KREW}.tar.gz"
+    rm -f "${KREW}.tar.gz"
+    ./"${KREW}" install krew
+
+    # Add krew to PATH by appending this line to .bashrc
+    krew_path_bashrc="export PATH=${KREW_ROOT:-${HOME}/.krew}/bin:${PATH}"
+    # Add the line if it is not already there
+    grep -qxF "${krew_path_bashrc}" ~/.bashrc || echo "${krew_path_bashrc}" >> ~/.bashrc
+    popd || return 1
+}
+
+download_and_install_minikube()
+{
+    MINIKUBE_URL="https://storage.googleapis.com/minikube/releases/${MINIKUBE_VERSION}/minikube-linux-amd64"
+    MINIKUBE_BINARY="minikube"
+
+    wget_and_verify "${MINIKUBE_URL}" "${MINIKUBE_SHA256}" "${MINIKUBE_BINARY}"
+    chmod +x "${MINIKUBE_BINARY}"
+    sudo mv "${MINIKUBE_BINARY}" /usr/local/bin/
+}
+
+download_and_install_kvm2_driver()
+{
+    DRIVER_URL="https://storage.googleapis.com/minikube/releases/${MINIKUBE_VERSION}/docker-machine-driver-kvm2"
+    DRIVER_BINARY="docker-machine-driver-kvm2"
+
+    wget_and_verify "${DRIVER_URL}" "${MINIKUBE_DRIVER_SHA256}" "${DRIVER_BINARY}"
+    chmod +x "${DRIVER_BINARY}"
+    sudo mv "${DRIVER_BINARY}" /usr/local/bin/
+}
+
+download_and_install_kind()
+{
+    KIND_URL="https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/kind-$(uname)-amd64"
+    KIND_BINARY="kind"
+
+    wget_and_verify "${KIND_URL}" "${KIND_SHA256}" "${KIND_BINARY}"
+    chmod +x "${KIND_BINARY}"
+    sudo mv "${KIND_BINARY}" /usr/local/bin/
+}
+
+download_and_install_tilt()
+{
+    # TODO: see if there is actual releases instead of just git sha
+    TILT_URL="https://raw.githubusercontent.com/tilt-dev/tilt/${TILT_VERSION}/scripts/install.sh"
+    TILT_SCRIPT="install.sh"
+
+    wget_and_verify "${TILT_URL}" "${TILT_SHA256}" "${TILT_SCRIPT}"
+    bash "${TILT_SCRIPT}"
+    rm "${TILT_SCRIPT}"
+}
+
+# FIXME: kubectl cannot be latest, use kubernetes version - 1
```

Comment:
Was this already fixed or am I missing something? :thinking:

Reply:
It is now taking same k8s version (1.27.1), but that is issue for upgrade etc, where the source is 1.25, which is beyond kubectl allowed version skew. So we should actually be installing kubectl 1.26 (hence "version - 1").

Reply:
Ah got it. :+1:

Reply:
I'll actually just leave it as TODO in the script, and fix is separately.